framework-support/jobrunr-spring-boot-2-starter/src/main/java/org/jobrunr/scheduling/AsyncJobPostProcessor.java

```diff
@@ -0,0 +1,49 @@
+package org.jobrunr.scheduling;
+
+import org.jobrunr.jobs.annotations.Job;
+import org.jobrunr.jobs.annotations.AsyncJob;
+import org.springframework.aop.framework.ProxyFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.stereotype.Component;
+
+import java.lang.reflect.InvocationTargetException;
+
+@Component
+public class AsyncJobPostProcessor implements BeanPostProcessor, BeanFactoryAware {
+    BeanFactory beanFactory;
+
+    @Override
+    public Object postProcessBeforeInitialization(Object bean, String beanName) {
+        if (bean.getClass().isAnnotationPresent(AsyncJob.class)) {
+            try {
+                return applyJobEnhancement(bean);
+            } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        return bean;
+    }
+
+    public Object applyJobEnhancement(Object bean) throws InstantiationException, IllegalAccessException, NoSuchMethodException, InvocationTargetException {
+        JobScheduler jobScheduler = beanFactory.getBean(JobScheduler.class);
+        ProxyFactory proxyFactory = new ProxyFactory(bean);
+        proxyFactory.setProxyTargetClass(true);
+        proxyFactory.addAdvice((org.aopalliance.intercept.MethodInterceptor) invocation -> {
```

Comment:
Why not just implement `MethodInterceptor` in the `JobInterceptor`? You would only need one single instance of the interceptor anyway. This would apply for the Spring Boot 3 one as well (wouldn't there be a way to share this code instead of duplicating it?).

Reply:
I fixed the first issue, but I am unsure about the second part about duplicating code followed the same way as it is done in other places.
@rdehuyss your input can help here.